src/metabase/query_processor/util/add_alias_info.clj

```diff
@@ -316,6 +321,11 @@
                        (qp.store/->legacy-metadata field)))
     (:name field)))
 
+(defn- field-requires-original-field-name
+  "JSON extraction fields need to be named with their outer `field-name`, not use any existing `::desired-alias`."
+  [field-clause]
+  (boolean (some-> field-clause field-instance :nfc-path)))
```

Comment:
Is there a test for this situation?

Reply:
Yes, but unfortunately they're DB-specific because JSON unfolding support is not widespread.